internal/operators/oai/oai_operator.go

```diff
@@ -0,0 +1,263 @@
+package oai
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/conversions"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	operatorName      = "oai"
+	operatorNamespace = "redhat-ods-operator"
+	operatorFullName  = "OpenShift AI"
+	subscriptionName  = "rhods-operator"
+)
+
+// operator is an OpenShift AI OLM operator plugin.
+type operator struct {
+	log    logrus.FieldLogger
+	config *Config
+}
+
+// NewOperator creates new OpenShift AI operator.
+func NewOperator(log logrus.FieldLogger) *operator {
+	config := &Config{}
+	err := envconfig.Process(common.EnvConfigPrefix, config)
```

Comment:
This happens in main as well

Reply:
Yes, but in "main" we don't load the environment variables that are loaded here. All the other operators use this same mechanism. We could probably add to the "Options" struct used in "main" these variables, but then we would need to find a way to pass those options to this "NewOperator" function. I'd prefer to not do that in this patch.